lib/Twig/Node/Expression/Call.php

```diff
@@ -287,6 +300,93 @@ private function reflectCallable($callable)
 
         return $this->reflector = array($r, $callable);
     }
+
+    public function compileInline(Twig_Compiler $compiler, $callable)
+    {
+        $args = array();
+
+        if ($this->hasAttribute('needs_environment') && $this->getAttribute('needs_environment')) {
+            $args[] = $compiler->getEnvironment();
+        }
+
+        if ($this->hasAttribute('needs_context') && $this->getAttribute('needs_context')) {
+            $args[] = array(); // @TODO: Is it possible to get the context from here?
+        }
+
+        if ($this->hasAttribute('arguments')) {
+            foreach ($this->getAttribute('arguments') as $argument) {
+                $args[] = sprintf('"%s"', addcslashes($argument, "\0\t\"\$\\"));
+            }
+        }
+
+        if ($this->hasNode('node')) {
+            $node = $this->getNode('node');
+
+            $args[] = $this->compileInlineArgument($compiler, $node);
+        }
+
+        if ($this->hasNode('arguments')) {
+            $callable = $this->getAttribute('callable');
+            $arguments = $this->getArguments($callable, $this->getNode('arguments'));
+            foreach ($arguments as $node) {
+                $args[] = $this->compileInlineArgument($compiler, $node);
+            }
+        }
+
+        $result = $callable(...$args);
+
+        if (is_string($result)) {
+            $compiler->string($result);
+        } else {
+            $compiler->raw(var_export($result, true));
+        }
+    }
+
+    private function compileInlineArgument(Twig_Compiler $compiler, Twig_Node $node)
+    {
+        if (!$node instanceof Twig_Node_Expression_Call && !$node instanceof Twig_Node_Expression_Constant && !$node instanceof Twig_Node_Expression_Array) {
+            throw new RuntimeException('Unsupported dynamic value: ' . $node);
+        }
+
+        if ($node instanceof Twig_Node_Expression_Filter) {
+            $name = $node->getNode('filter')->getAttribute('value');
+            $filter = $compiler->getEnvironment()->getFilter($name);
+
+            return $node->compileInline($compiler, $filter->getCallable());
```

Comment:
this is broken, as `compileInline` does not return anything

Reply:
Thanks, this explains why I couldn't get it to work on a some of the core filters, since I was getting a lot of `null` values :)